snuba/query/composite.py

```diff
@@ -62,6 +62,30 @@ def __init__(
         )
         self.__from_clause = from_clause
 
+    def __repr__(self) -> str:
+        from snuba.query.formatters.tracing import format_query
```

Comment:
I would assume that the import inline is here because of a circular dependency. What is the issue specifically?
Generally when we have to add an import inline because of a circular dependency is because we did not organize the code properly and we should try to fix it instead.

Reply:
tracing formatter needs the Query types for typing, query needs the formatter for repr.

Reply:
Ok, I tried that and it opens up a whole can of worms due to the `JoinVisitor` which needs `TSimpleDataSource` w and `CompositeQuery`. A lot of those are just for type annotations as well BUT if you use the `TYPE_CHECKING` thing then the types have to be written as strings which makes this:

```python3
class IndividualNode(JoinNode[TSimpleDataSource], Generic[TSimpleDataSource])
```

into this:

```python3
class IndividualNode(JoinNode['TSimpleDataSource'], Generic['TSimpleDataSource'])
```

which is not valid syntax. If we were on python3.10 it would work but it does not